subunit.go

```diff
@@ -68,26 +70,32 @@ type StreamResultToBytes struct {
 	Output io.Writer
 }
 
-type packet struct {
-	testID string
-	status string
+// Event is a status or a file attachment event.
+type Event struct {
```

Comment:
IMO it would be better if the production code is not aware of the tests, if it doesn't have different execution paths for the different environments. Maybe an interface here could be useful, with different implementations for production and for testing, what do you think?